### PR TITLE
[ML-DataFrame] Remove ID field from data frame indexer stats

### PR DESCRIPTION
**Pull request by droberts195**
This is a followup to #44350. The indexer stats used to
be persisted standalone, but now are only persisted as
part of a state-and-stats document. During the review
of #44350 it was decided that we'll stick with this
design, so there will never be a need for an indexer
stats object to store its transform ID as it is stored
on the enclosing document. This PR removes the indexer
stats document ID.

### CODE REVIEW

**Comment by droberts195** [comment]
Jenkins run elasticsearch-ci/2

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameIndexerTransformStats.java [hunk 1]
```diff
@@ -57,60 +55,46 @@
         LENIENT_PARSER.declareLong(constructorArg(), SEARCH_FAILURES);
     }
 
-    private final String transformId;
-
     /**
      * Certain situations call for a default transform ID, e.g. when merging many different transforms for statistics gather.
```
Comment (hendrikmuhs) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameIndexerTransformStats.java]: nit: these comments aren't true anymore

**Review by hendrikmuhs — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameIndexerTransformStats.java (hunk 1 above)
Comment (droberts195) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameIndexerTransformStats.java]: I removed the method as well as the comment as after the other changes I don't think it was adding much over calling the default constructor.

**Pull request merged**